src/utils/helpers.ts

```diff
@@ -165,3 +166,46 @@ export const groupAndFilterConversations = (
 
   return groupedFilteredResults;
 };
+
+const removeSystemMessages = (messages: ChatMessage[]) => {
+  return messages.filter((msg) => msg.role !== 'system');
+};
+
+export const formatContentSnippet = (
+  conversation: SearchableChatHistory,
+  searchTerm: string = '',
+): string => {
+  const messages = removeSystemMessages(conversation.messages);
+
+  if (searchTerm.trim() !== '') {
+    for (const message of messages) {
+      const content = extractTextContent(message);
+      const searchRegex = new RegExp(searchTerm, 'i');
+      const match = content.match(searchRegex);
+
+      //  If a match is found
+      if (match && match.index !== undefined) {
+        const searchIndex = match.index;
+        //  Find start of snippet considering up to 3 words before match
+        let snippetStart = searchIndex;
+        if (searchIndex > 0) {
+          const beforeMatch = content.slice(0, searchIndex).trim();
+          const precedingWords = beforeMatch.split(' ').slice(-3);
+          snippetStart = searchIndex - (precedingWords.join(' ').length + 1);
+          if (snippetStart < 0) snippetStart = 0;
+        }
+
+        return content.slice(snippetStart, snippetStart + 100).trim();
+      }
+    }
+  }
+
+  //  Fallback to the first user message if no search term or no matches
+  const firstUserMessage = messages.find((msg) => msg.role === 'user');
+  if (firstUserMessage) {
+    const content = extractTextContent(firstUserMessage);
+    return content.slice(0, 100);
+  }
```

Comment:
If the match is in the title and not the content, then just show the beginning of the first user's message.